src/main/kotlin/com/shankhadeepghoshal/kotlinfinaltask/pojo/Entities.kt

```diff
@@ -0,0 +1,30 @@
+package com.shankhadeepghoshal.kotlinfinaltask.pojo
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.shankhadeepghoshal.kotlinfinaltask.config.DogDeserializer
+import org.springframework.data.annotation.Id
+import org.springframework.data.relational.core.mapping.Table
+
+@Table(name = "dog")
+data class Dog(
+    @Id val id: Int? = null,
+    val nameMain: String,
+    val subName: String?
+) {
+    constructor(nameMain: String, subName: String?) : this(null, nameMain, subName)
```

Comment:
This constructor is redundant. Since you are using default value for id, this parameter can be omitted when creating an object. You can use named arguments to achieve the same result:
```
Dog(nameMain = <your value>, subName = <your value>)
```